src/app/qgsvectorlayerproperties.cpp

```diff
@@ -1235,16 +1235,21 @@ void QgsVectorLayerProperties::addJoinToTreeWidget( const QgsVectorLayerJoinInfo
     joinItem->setText( 3, QChar( 0x2714 ) );
   }
 
-  joinItem->setText( 4, join.prefix() );
+  if ( join.isDynamicFormEnabled() )
+  {
+    joinItem->setText( 4, QChar( 0x2714 ) );
```

Comment:
Note to self, maybe we could have some #define or enum for unicode chars used in the UI